extensions/ql-vscode/src/remote-queries/variant-analysis-monitor.ts

```diff
@@ -91,6 +86,57 @@ export class VariantAnalysisMonitor extends DisposableObject {
     return { status: 'CompletedSuccessfully', scannedReposDownloaded: scannedReposDownloaded };
   }
 
+  private async scheduleForDownload(
+    scannedRepo: VariantAnalysisScannedRepository,
+    variantAnalysisSummary: VariantAnalysisApiResponse
+  ): Promise<void> {
+    void commands.executeCommand('codeQL.autoDownloadVariantAnalysisResult', scannedRepo, variantAnalysisSummary);
+  }
+
+  private shouldDownload(
+    scannedRepo: VariantAnalysisScannedRepository,
+    alreadyDownloaded: number[]
+  ): boolean {
+    return (!alreadyDownloaded.includes(scannedRepo.repository.id) && scannedRepo.analysis_status === 'succeeded');
+  }
+
+  private getReposToDownload(
+    variantAnalysisSummary: VariantAnalysisApiResponse,
+    alreadyDownloaded: number[]
+  ): VariantAnalysisScannedRepository[] {
+    if (variantAnalysisSummary.scanned_repositories) {
+      return variantAnalysisSummary.scanned_repositories.filter(scannedRepo => this.shouldDownload(scannedRepo, alreadyDownloaded));
+    } else {
+      return [];
+    }
+  }
+
+  private async downloadVariantAnalysisResults(
+    variantAnalysisSummary: VariantAnalysisApiResponse,
+    scannedReposDownloaded: number[]
+  ): Promise<number[]> {
+    const repoResultsToDownload = this.getReposToDownload(variantAnalysisSummary, scannedReposDownloaded);
+    const downloadedRepos: number[] = [];
+
+    const batchSize = 3;
+    const numOfBatches = Math.ceil(repoResultsToDownload.length / batchSize);
+
+    for (let i = 0; i < repoResultsToDownload.length; i += batchSize) {
+      const batch = repoResultsToDownload.slice(i, i + batchSize);
+
+      const nwos = batch.map(s => s.repository.full_name).join(', ');
+      const batchTasks = batch.map(scannedRepo => this.scheduleForDownload(scannedRepo, variantAnalysisSummary));
+
+      void this.logger.log(`Downloading batch ${Math.floor(i / batchSize) + 1} of ${numOfBatches} (${nwos})`);
+
+      await Promise.allSettled(batchTasks);
+
+      batch.map(scannedRepo => downloadedRepos.push(scannedRepo.repository.id));
```

Comment:
I think this can be simplified to:

```suggestion
      downloadRepos.push(...batch.map(scannedRepo => scannedRepo.repository.id));
```

This will also remove the `map` call which we're not using (if that's what we wanted to do, it should probably have been `forEach`).